src/rbac/roleManager.ts

```diff
@@ -13,7 +13,11 @@
 // limitations under the License.
 
 // RoleManager provides interface to define the operations for managing roles.
+import { MatchingFunc } from './internalRoleManager';
+
 export interface RoleManager {
+  addDomainMatchingFunc(fn: MatchingFunc): Promise<void>;
```

Comment:
you should make this to be optional, and return Promise<T> | T, so like:
```
addDomainMatchingFunc?(fn: MatchingFunc): Promise<void> | void;
addMatchingFunc?(fn: MatchingFunc): Promise<void> | void
```
other methods are the same.